drivers/sensor/tmag5273/tmag5273.h

```diff
@@ -0,0 +1,487 @@
+
+#define TMAG5273_INTB_PIN_LOW			0
+#define TMAG5273_INTB_PIN_HIGH			1
+#define TMAG5273_OSC_ERR			1
+#define TMAG5273_INT_ERR			1
+#define TMAG5273_OTP_CRC_ERR			1
+#define TMAG5273_VCC_UV_ERR			1
+
+/* additional values */
+#define TMAG5273_MEAS_RANGE_LOW_MT_VER1		40
+#define TMAG5273_MEAS_RANGE_HIGH_MT_VER1	80
+#define TMAG5273_MEAS_RANGE_LOW_MT_VER2		133
+#define TMAG5273_MEAS_RANGE_HIGH_MT_VER2	266
+
+#define TMAG5273_TEMPERATURE_T_SENS_T0		25
+#define TMAG5273_TEMPERATURE_T_ADC_T0		17508
+#define TMAG5273_TEMPERATURE_T_ADC_RES		60.1
+
+#define TMAG5273_T_START_SLEEP_US		50
+#define TMAG5273_T_GO_SLEEP_US			20
+
+/** OR this bit to any register address to trigger a conversion in standby mode */
+#define TMAG5273_CONVERSION_START_BIT		0x80
+
+/* -------------------------------------------------------------------- */
+
+#define TMAG5273_I2C_BUFFER_SIZE (TMAG5273_REG_RESULT_END - TMAG5273_REG_RESULT_BEGIN + 2)
+
+/* -------------------------------------------------------------------- */
+
+/* --- helper macros --- */
+
+/**
+ * @brief   Macro to set bits in a register \a reg_data
+ *
+ * Mask and position are automatically resolved.
+ *
+ * @param[in]  reg_data    \c uint8_t register value
+ * @param[in]  bit_name    Macro name of the bit. See example for details.
+ * @param[in]  data        Data which is used to set the bits
+ *
+ * @return  updated \a reg_data
+ *
+ * @par  Example
+ * The following example uses (internally) the definitions \c TMAG5273_INT_STATE_MSK
+ * and \c TMAG5273_INT_STATE_POS to set the bits of the register \c INT_CONFIG_1.
+ *    @code{.c}
+ *       uint8_t  reg_int_config_1 = 0x03; // initialized to some value
+ *       reg_int_config_1 = TMAG5273_SET_REG_BITS(reg_int_config_1,
+ *           TMAG5273_INT_STATE, TMAG5273_INT_STATE_PULSE);
+ *    @endcode
+ */
+#define TMAG5273_SET_REG_BITS(reg_data, bit_name, data) \
+	((reg_data & ~(bit_name##_MSK)) | \
+	 ((data << bit_name##_POS) & bit_name##_MSK))
+
+/**
+ * @brief   Macro to get bits frm a register \a reg_data
+ *
+ * @param[in]  reg_data    \c uint8_t register value
+ * @param[in]  bit_name    Macro name of the bit
+ *
+ * @return  bit value of \a bit_name
+ *
+ * @note
+ *    Returned value reflect the bits \b without taking the position into account.
+ *    For example bit \c INT_STATE in register \c INT_CONFIG_1 : \n
+ *
+ *    |      0    |      1   | 2 | 3 | 4 |    5      |    6       |     7    |
+ *    |:---------:|:--------:|:-:|:-:|:-:|:---------:|:----------:|:--------:|
+ *    | MASK_INTB | RESERVED | - | - | - | INT_STATE | THRSLD_INT | RSLT_INT |
+ *
+ *    @code{.c}
+ *       // preparation: set bit at position INT_STATE
+ *       uint8_t  acc_config0 = b11 << TMAG5273_INT_STATE_POS;
+ *       // readout the INT_STATE bits using the macro
+ *       TMAG5273_GET_REG_BITS(acc_config0, BMA400_OSR_LP); // b11
+ *    @endcode
+ */
+#define TMAG5273_GET_REG_BITS(reg_data, bit_name) \
+	((reg_data & (bit_name##_MSK)) >> (bit_name##_POS))
```

Comment:
Just use `include/zephyr/sys/util.h` helper macros directly instead

Reply:
`TMAG5273_SET_REG_BITS` is needed, since it does both `FIELD_PREP` as well as clearing the bit in prior to the logical or. I changed the naming to make it more streamlined.